src/plugin.cpp

```diff
@@ -1,36 +1,94 @@
 #include <orthanc/OrthancCPlugin.h>
+#include <string>
+#include <regex>
 #include "configuration.h"
 
-static OrthancPluginContext* context_ = nullptr;
+static OrthancPluginContext *context_ = nullptr;
 
-extern "C" {
-    int32_t OrthancPluginInitialize(OrthancPluginContext* context){
-        context_ = context;
-
-        /* Check the version of the Orthanc core */
-        if (OrthancPluginCheckVersion(context) == 0){
-            char info[256];
-            sprintf(info, "Your version of Orthanc (%s) must be above %d.%d.%d to run this plugin",
-                    context->orthancVersion,
-                    ORTHANC_PLUGINS_MINIMAL_MAJOR_NUMBER,
-                    ORTHANC_PLUGINS_MINIMAL_MINOR_NUMBER,
-                    ORTHANC_PLUGINS_MINIMAL_REVISION_NUMBER);
-
-            OrthancPluginLogError(context, info);
-            return -1;
-        }
-        return 0;
-    }
-    
-    void OrthancPluginFinalize(){
+OrthancPluginErrorCode OnStoredCallback(OrthancPluginDicomInstance *instance, const char *instanceId) {
+    char buffer[1024];
+    sprintf(buffer, "Just received a DICOM instance of size %d and ID %s from origin %d (AET %s)",
+            (int) OrthancPluginGetInstanceSize(context_, instance), instanceId,
+            OrthancPluginGetInstanceOrigin(context_, instance),
+            OrthancPluginGetInstanceRemoteAet(context_, instance));
+    OrthancPluginLogWarning(context_, buffer);
 
-    }
+    std::string instanceId_str = std::string(instanceId);
+
+    OrthancPluginErrorCode error;
+
+    OrthancPluginMemoryBuffer anonymized_buffer;
+    std::string anonymize_uri = "/instances/" + instanceId_str + "/anonymize";
```

Comment:
So the API orthanc provides does need the images to be stored already. It looks like it relies on the instance id, which we only have because of the on stored callback.